content/file/utils.go

```diff
@@ -180,15 +181,27 @@ func extractTarDirectory(dir, prefix string, r io.Reader, buf []byte) error {
 		case tar.TypeDir:
 			err = os.MkdirAll(path, header.FileInfo().Mode())
 		case tar.TypeLink:
+			// NOTE: ORAS does not generate hard links when creating tarballs.
+			// If a hard link is found in the tarball, it will be extracted.
+			// If the target link already exists, os.Link will throw an error.
+			// This is a known limitation and will not be addressed.
```

Comment:
If supporting links is possible and maybe desirable in the futre, maybe just make an issue out of this

Reply:
Hard links in tarballs are tricky to support and can potentially introduce issues. For example, it requires exhaustive handling/tests on different OS. If there is a valid use case, we can evaluate the scenario and see if we need an issue for this.